### DISCUSSION TITLE
Adding options to us local LLM's like GPT4all, or alpaca

### DISCUSSION THREAD

**Discussion by CreeperNinjaYT** [brianpetro/obsidian-smart-connections#141] [Ideas]
I think it would be fantastic if the Smart Connections plugin could utilize a local language model (LLM) on our own computers for searching and smart connections. 
One idea is to incorporate the alpaca.cpp program into Smart Connections and provide a way for users to direct the plugin to the LLM they've downloaded on their own PC. 
This would give users the flexibility to switch to more advanced LLMs as they become available, without requiring significant changes in the code of the plugin.

**Comment by brianpetro** [comment]
@CreeperNinjaYT see https://github.com/brianpetro/obsidian-smart-connections/issues/137#issuecomment-1492775955
